lib/init.g

```diff
@@ -658,6 +658,35 @@ DeclareUserPreference( rec(
   default:= 3,
   check:= val -> IsInt( val ) and 0 <= val,
   ) );
+DeclareUserPreference( rec(
+  name:= "ReproducibleBehaviour",
+  description:= [
+    "This preference disables code in GAP which changes behaviour based on time \
+spent, and therefore can produce different results depending on how much time is \
+taken by other programs running on the same computer. This option may lead to \
+slower or lower-quality results.\
+\n\
+Note that many algorithms in GAP use the global random number generator, which is \
+NOT affected by this option. This only tries to ensure the same version of GAP, \
+with the same package versions loaded, on the same machine, running the same code, \
+in a fresh GAP session, will produce the same results."
+    ],
+  default:= false,
+  values:= [ true, false ],
+  multi:= false,
+  ) );
+
+# This provides a subtitute for the function Runtime(), for use
+# with the user option ReproducibleBehaviour. Instead of measuring
+# time, it simply increments a counter each time it is called.
+FAKE_RUNTIME := (function()
+                   local tick;
+                   tick := 0;
+                   return function()
+                        tick := tick + 1;
+                        return tick;
+                   end;
+                 end)();
```

Comment:
I dislike this us of a closure to keep an internal global counter. It makes the code slightly harder to understand, IMHO; but a more serious issue is that it makes it impossible to reset the timer other than by resetting it.

For bug fixing purposes, on several occassions the only way I could create a reproducible test case was to run an example which sometimes triggered a problem, say a few thousand times, each time saving the RNG state just beforehand; and once I hit the error, I could use the stored RNG to reliably reproduce the issue. Without that ability, I'd not have been able to fix several bugs in `recog`.

I imagine that debugging code using a timer will require similar techniques, so saving and restoring the fake timer state seems like a very important ability to me.

So I'd greatly prefer if that state could be stored in an explicit global variable.